### PR TITLE
PWA

### PR DESCRIPTION
**Pull request by axeldelafosse**
# Why

Most of our traffic is on web at the moment, so we need to let users install the web app as a PWA and use a service worker to improve the performances.

# How

- Installed `next-pwa`. Used it in the past, it's pretty good! The default runtime cache works for us: https://github.com/shadowwalker/next-pwa/blob/master/cache.js
- Added a redirection to the stores at `/app-store` and `/google-play`
- Added preloading and DNS prefetching for our API and CDN

# Test Plan

Check if the service worker is working correctly with your browser dev tools

### CODE REVIEW

File: apps/next/src/pages/_app.tsx [hunk 1]
```diff
@@ -94,13 +94,33 @@ export default function App({ Component, pageProps, router }: AppProps) {
   return (
     <>
       <Head>
+        {process.env.NODE_ENV !== "development" ? (
+          <>
+            <link rel="preconnect" href="//showtimenft.wl.r.appspot.com" />
+            <link rel="dns-prefect" href="//showtimenft.wl.r.appspot.com" />
+            <link rel="preconnect" href="//lh3.googleusercontent.com" />
+            <link rel="dns-prefect" href="//lh3.googleusercontent.com" />
+          </>
```
Comment (intergalacticspacehighway) [inline apps/next/src/pages/_app.tsx:103]: 💯

**Review by alantoa — APPROVED** [review]
just checked it, working fine for me! 
also maybe we can share to [here](https://github.com/shadowwalker/next-pwa/discussions/206) later. 
🔥 🔥 🔥

File: apps/next/workbox-cache.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      },
+      networkTimeoutSeconds: 10, // fall back to cache if api does not response within 10 seconds
+    },
+  },
+  {
+    urlPattern: ({ url }) => {
+      const isSameOrigin = self.origin === url.origin;
+      if (!isSameOrigin) return false;
+      const pathname = url.pathname;
+      if (pathname.startsWith("/api/")) return false;
```
Comment (intergalacticspacehighway) [inline apps/next/workbox-cache.js:155]: nit: nice 💯 later we can remove the swr cache on web

**Review by intergalacticspacehighway — APPROVED** [review]
LGTM ❇️

**Pull request merged**